### DISCUSSION TITLE
Tiempo de ejecución de tests

### DISCUSSION THREAD

**Discussion by SkylerEverett** [IIC2133-PUC/2024-2#22] [Tarea 0] [answered]
Hola,

La tarea dice:
> "Para cada test de evaluación, tu programa deberá entregar el output correcto en menos de 5 segundos y utilizar menos de 1 GB de  RAM. De lo contrario, recibirás 0 puntos en ese test."

Me pasa que, por los prints puestos en consola para debugear e ir analizando el proceso de algunos casos, cuando se ejecuta el programa normalmente con `./puffle input.txt output.txt` con un caso hard, toma más de 5 segundos en ejecutarse como tal. Sin embargo, creé un script que me permite comparar los outputs para ver si el código está bien hecho y este omite la salida de la consola. Solo ejecuta el programa como tal.

```
Part-1-hard...
Took time: 0m0.003s
==> ✅ PASSED
Part-2-hard-1...
Took time: 0m0.309s
==> ✅ PASSED
Part-2-hard-2...
Took time: 0m0.280s
==> ✅ PASSED
```

En este caso, los programas se ejecutan super rápido. Considerando cómo harán la recorrección, me preguntaba si debía comentar las líneas que hacen print o si lo entrego de esta manera.

Quedo atento a la respuesta :)
Desde ya, gracias 🫰🏻

**Comment by tomastrivino** [comment, marked as answer]
Hola!
Te recomendamos comentar las líneas que hacen `print` **en la consola** (printf), ya que ralentizan el programa y, al final, lo que se compara es tu output en el archivo. 
Como se menciona en el enunciado, tu código debe ejecutarse en menos de 5 segundos. Si tu test `hard-2` toma 6 segundos, será evaluado con 0 puntos (ojo que esto es por tests, por lo que si `hard-1` toma menos de 5 segundos y está correcto, tendrás el puntaje completo para dicho test).
